publisher-sdk/src/main/java/com/criteo/publisher/Criteo.java

```diff
@@ -104,8 +105,11 @@ private static Criteo init(
           dependencyProvider.setCriteoPublisherId(criteoPublisherId);
 
           DeviceUtil deviceUtil = dependencyProvider.provideDeviceUtil();
+          AndroidUtil androidUtil = dependencyProvider.provideAndroidUtil();
 
-          if (deviceUtil.isVersionSupported()) {
+          if (!androidUtil.isCurrentProcessInForeground(application)) {
+            criteo = null;
```

Comment:
This make this method nullable. Hence it is not safe anymore to do init+getinstance. Maybe could you use the dummy criteo instead ?

Reply:
The dummy criteo instance wouldn't work. You could have a case where Criteo is initialized from a background service, then initialized properly from a foreground process. When this happens, the normal initialization won't happen as criteo would already be initialized and stuck on `DummyCriteo` until next process restart. If the service always call `Application#onCreate` first, then Criteo would never initialize properly.
We could make it work by changing the condition here: https://github.com/criteo/android-publisher-sdk/blob/main/publisher-sdk/src/main/java/com/criteo/publisher/Criteo.java#L100, to be: 

`if (criteo == null || criteo instanceof DummyCriteo)`

Reply:
Oh ok, I wasn't aware of this. Ok to change the condition then. Please also add a comment explaining the why of such condition.
Moreover, could you add a test validating the scenario you described ?